src/containers/Application/ApplicationOverview.tsx

```diff
@@ -71,13 +83,44 @@ const ApplicationOverview: React.FC = () => {
     }
   }, [elementsState, responsesLoading])
 
+  const revalidateAndUpdate = async () => {
+    const revalidate = await revalidateAll(elementsState, responsesByCode)
+
+    // Update database if validity changed
+    revalidate.validityFailures.forEach((changedElement) => {
+      responseMutation({
+        variables: {
+          id: changedElement.id,
+          isValid: changedElement.isValid,
+        },
+      })
+    })
+
+    // If invalid responses, re-direct to first invalid page
+    if (!revalidate.allValid) {
+      console.log('Some responses invalid')
+      const { firstErrorSectionIndex, firstErrorPage } = getFirstErrorLocation(
+        revalidate.validityFailures,
+        elementsState as ApplicationElementStates
+      )
+      // TO-DO: Alert user of Submit failure
+      push(`/application/${serialNumber}/S${firstErrorSectionIndex + 1}/Page${firstErrorPage}`)
```

Comment:
Ahh, I know this might require quite a big change, but we can't rely on the code being the same as the `index + 1`. This should be the section code instead. If `elementState` doesn't have the `sectionCode` perhaps is better to change it to have it, or we can change the URL to always use the sectionIndex, that could also be done.

Reply:
Although since the index start from 0 it might be weird on the URL

Reply:
Oh, my bad, I thought the URL was just using "S" + sectionIndex. Yes, I'll fix this for sure.

Reply:
Done 682b09d :)